### PR TITLE
Added support for custom scene loading and HERE SDK 4.13.3.0

### PR DESCRIPTION
**Pull request by tstaszak89**
Application version increased to 1.5.1
Added file_picker plugin.
Application is now compatible with HERE SDK 4.13.3.0
Disabled bitcode (iOS) because it is deprecated by Apple. HERE SDK starting from 4.13.3.0 no longer includes bitcode.

### CODE REVIEW

File: lib/common/custom_map_style_settings.dart [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  String? get customMapStyleFilepath => _customMapStyleFilepath;
+
+  /// Setter for custom map style filepath.
+  void set customMapStyleFilepath(String? newFilepath) {
+    _customMapStyleFilepath = newFilepath;
+    notifyListeners();
+  }
+
+  /// Returns filename of current custom map style. Empty string is returned when custom map style is not set.
+  String getCustomMapStyleFilename() {
```
Comment (ppiatkowski-here) [inline lib/common/custom_map_style_settings.dart:36]: (0) Seems like a good situation to use a getter:
```
String get customMapStyleFilename {
   //...
}
```
Decision is up to you. Both versions work.

File: lib/navigation/navigation_screen.dart [hunk 2]
```diff
@@ -230,7 +233,17 @@ class _NavigationScreenState extends State<NavigationScreen> with WidgetsBinding
...
+
+    customStyleSettings.customMapStyleFilepath != null
+        ? hereMapController.mapScene.loadSceneFromConfigurationFile(
+            customStyleSettings.customMapStyleFilepath!,
+            mapLoadSceneCallback,
+          )
+        : hereMapController.mapScene.loadSceneForMapScheme(
+            MapScheme.normalDay,
+            mapLoadSceneCallback,
+          );
```
Comment (ppiatkowski-here) [inline lib/navigation/navigation_screen.dart:246]: This snippet is repeated 4 times in different files. Can you make it a function taking hereMapController, customStyleSettings and mapLoadSceneCallback as parameters?

File: lib/common/custom_map_style_settings.dart (hunk 1 above)
Comment (tstaszak89) [reply lib/common/custom_map_style_settings.dart]: You are right, I changed it into a getter.

File: lib/navigation/navigation_screen.dart (hunk 2 above)
Comment (tstaszak89) [reply lib/navigation/navigation_screen.dart]: Yes, I created a helper function for this.

**Review by ppiatkowski-here — APPROVED** [review]

**Review by mohan-manthri — APPROVED** [review]

**Review by Guru-cm-here — APPROVED** [review]

**Review by ppiatkowski-here — APPROVED** [review]

**Review by ppiatkowski-here — APPROVED** [review]

**Pull request merged**